Initial Commit

JorgePatrick opened the pull request:
Sum 1 to 3 in Roman

JorgePatrick committed:
Initial Commit

Sum 1 to 3 in Roman

chicofaccin commented on src/main/java/com/jorgepatrick/RomanCalculator.java:
```diff
@@ -0,0 +1,55 @@
+
+        int firstArabicNumber = 0;
+        int secondArabicNumber = 0;
+        firstArabicNumber = parseRomanToArabic(firstAddend);
```
You can declare the `firstArabicNumber ` and `secondArabicNumber ` as final. So they would be immutable. 

```
final int firstArabicNumber = parseRomanToArabic(firstAddend);
final int secondArabicNumber = parseRomanToArabic(secondAddend);
```

chicofaccin commented on src/main/java/com/jorgepatrick/RomanCalculator.java:
```diff
@@ -0,0 +1,55 @@
+        return parseArabicToRoman(arabicNumberSum);
+    }
+
+    private String parseArabicToRoman(int arabicNumberSum) {
```
`private String parseArabicToRoman(final int arabicNumberSum) {`

chicofaccin commented on src/main/java/com/jorgepatrick/RomanCalculator.java:
```diff
@@ -0,0 +1,55 @@
+        }
+
+        if (arabicNumberSum > 5) {
+            romanNumber = "V";
```
You can create a class and move all those constants to there. i.e.:
So you can reuse then and don't need to repeat it everywhere. 
````
public class RomanNumbers {
public static final String I = "I";
public static final String V = "V";
}
````

JorgePatrick committed:
Fixes as per PR

Modify variables to final
Create a class for constants (RomanNumbers)

chicofaccin commented on src/main/java/com/jorgepatrick/RomanCalculator.java:
```diff
@@ -0,0 +1,54 @@
+        }
+
+        for (int i = 0; i < unitQuantity; i++) {
+            romanNumber += 'I';
```
You can use the constant `I` here as well

JorgePatrick replied:
Havent used because itsnt exactly the roman number, it is the char 'I', should I,ve used the constant?

chicofaccin commented on src/main/java/com/jorgepatrick/RomanCalculator.java:
```diff
@@ -0,0 +1,54 @@
+        int unit = 0;
+
+        for (int i = 0; i < romanNumber.length(); i++) {
+            if (romanNumber.toUpperCase().charAt(i) == 'I') {
```
You can use the constant `I` here as well.

JorgePatrick replied:
Wasnt possible, cause the charAt requires a char and the constant is a string

chicofaccin replied:
You can do something like that -> `I.charAt(0)`

chicofaccin commented on src/main/java/com/jorgepatrick/RomanNumbers.java:
```diff
@@ -0,0 +1,10 @@
+    public static final String III = "III";
+    public static final String IV = "IV";
+    public static final String V = "V";
+    public static final String VI = "VI";
```
Maybe it should have only the symbols. Like `I` and `V`. Who is using them can compose the number.

JorgePatrick replied:
But wont that make it harder to read i.e.: Arguments.of(I + I, I, I + I + I)?

chicofaccin replied:
In the test, you don't necessarily need to use the constants. If it's easy, and it probably is, you can put the values

JorgePatrick committed:
Fix 2 PR

Use and concatenate Roman Symbols to create Roman Numbers
Change class from RomanNumbers to RomanSymbols
